src/e2e_test/perf_tool/pkg/metrics/recorder.go

```
@@ -44,14 +45,26 @@ func NewMetricsRecorder(pxCtx *pixie.Context, clusterCtx *cluster.Context, spec
+		recorderCtx := clusterCtx
+		ownsCtx := false
+		if promSpec.KubeconfigPath != "" || promSpec.KubeContext != "" {
+			var err error
+			recorderCtx, err = cluster.NewContextFromOptions(promSpec.KubeconfigPath, promSpec.KubeContext)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create cluster context for prometheus recorder: %w", err)
+			}
+			ownsCtx = true
 		}
+		return &prometheusRecorderImpl{
+			clusterCtx:    recorderCtx,
```

golangci-lint (2.12.2)

[error] File is not properly formatted

(gci)